File: java/javafind/lib/src/main/java/javafind/Color.java
package javafind;

public enum Color {
    BLACK("black"),
    RED("red"),
    GREEN("green"),
    YELLOW("yellow"),
    BLUE("blue"),
    MAGENTA("magenta"),
    CYAN("cyan"),
    WHITE("white");

    private final String name;

    Color(final String name) {
        this.name = name;
    }

    public static Color forName(final String name) {
        final var lName = name.trim().toLowerCase();
        for (final Color color : values()) {
            if (color.name.equals(lName)) {
                return color;
            }
        }
        return BLACK;
    }

    public ConsoleColor toConsoleColor() {
        return switch (this) {
            case BLACK -> ConsoleColor.BLACK;
            case RED -> ConsoleColor.RED;
            case GREEN -> ConsoleColor.GREEN;
            case YELLOW -> ConsoleColor.YELLOW;
            case BLUE -> ConsoleColor.BLUE;
            case MAGENTA -> ConsoleColor.MAGENTA;
            case CYAN -> ConsoleColor.CYAN;
            case WHITE -> ConsoleColor.WHITE;
        };
    }

    public String getName() {
        return this.name;
    }
}

File: kotlin/ktfind/lib/src/main/kotlin/ktfind/Color.kt
package ktfind

enum class Color(val value: String) {
    BLACK("black"),
    RED("red"),
    GREEN("green"),
    YELLOW("yellow"),
    BLUE("blue"),
    MAGENTA("magenta"),
    CYAN("cyan"),
    WHITE("white");

    fun toConsoleColor(): ConsoleColor {
        return when (this) {
            BLACK -> ConsoleColor.BLACK
            RED -> ConsoleColor.RED
            GREEN -> ConsoleColor.GREEN
            YELLOW -> ConsoleColor.YELLOW
            BLUE -> ConsoleColor.BLUE
            MAGENTA -> ConsoleColor.MAGENTA
            CYAN -> ConsoleColor.CYAN
            WHITE -> ConsoleColor.WHITE
        }
    }

    override fun toString(): String {
        return value
    }
}
